### PR TITLE
Use common webpack config

### PR DESCRIPTION
**Pull request by raimund-schluessler**
This PR introduces the common webpack config.

However, `npm run styleguide` is broken (see #3634), which needs to be fixed before we can decide how to proceed here.

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]

**Comment by raimund-schluessler** [comment]
Don't merge this yet, `npm run styleguide` is not happy about it:

```
> @nextcloud/vue@7.3.0 styleguide
> vue-styleguidist server

Building @nextcloud/vue 7.3.0 

This build version hash is 78efc74 

ValidationError: webpack Dev Server Invalid Options

options should NOT have additional properties
options should NOT have additional properties

ValidationError: webpack Dev Server Invalid Options

options should NOT have additional properties
options should NOT have additional properties

    at validateOptions (/home/raimund/nextcloud/apps2/nextcloud-vue/node_modules/vue-styleguidist/node_modules/webpack-dev-server/node_modules/schema-utils/src/validateOptions.js:32:11)
    at new Server (/home/raimund/nextcloud/apps2/nextcloud-vue/node_modules/vue-styleguidist/node_modules/webpack-dev-server/lib/Server.js:57:5)
    at createServer (/home/raimund/nextcloud/apps2/nextcloud-vue/node_modules/vue-styleguidist/lib/scripts/create-server.js:38:19)
    at server (/home/raimund/nextcloud/apps2/nextcloud-vue/node_modules/vue-styleguidist/lib/scripts/server.js:11:49)
    at Object.commandServer (/home/raimund/nextcloud/apps2/nextcloud-vue/node_modules/vue-styleguidist/lib/scripts/binutils.js:118:41)
    at runIt (/home/raimund/nextcloud/apps2/nextcloud-vue/node_modules/vue-styleguidist/lib/bin/styleguidist.js:78:16)
```

File: webpack.config.js [hunk 1]
```diff
@@ -22,151 +24,114 @@ const SCOPE_VERSION = JSON.stringify(versionHash)
...
+	'@nextcloud/calendar-js',
+	'tributejs',
+])
+
+webpackConfig.module.rules = Object.values(webpackRules)
+
+// Alias the default server assets path to link to local one
+webpackConfig.resolve.alias = {
+	'/core/css/../img': path.join(__dirname, 'styleguide/assets/img'),
+	'/core/css/../fonts': path.join(__dirname, 'styleguide/assets/fonts'),
```
Comment (skjnldsv) [inline webpack.config.js:109]: is this still needed?

Comment (raimund-schluessler) [reply webpack.config.js]: Seems to work fine without it. I removed it.

**Comment by raimund-schluessler** [comment]
> Don't merge this yet, `npm run styleguide` is not happy about it:

Fixed it, `npm run styleguide` works for me now.

**Review by GretaD — APPROVED** [review]

**Comment by skjnldsv** [comment]
Awesome @raimund-schluessler :hugs:

File: webpack.config.js [hunk 2]
```diff
@@ -22,151 +24,108 @@ const SCOPE_VERSION = JSON.stringify(versionHash)
...
+			.replace('src/mixins/', 'Mixins/')
+		acc[name] = path.join(__dirname, item)
+		return acc
+	}, {}),
+}
+
+webpackConfig.output = {
+	path: path.resolve(__dirname, './dist'),
+	publicPath: '/dist/',
+	filename: `[name].${libraryTarget}.js`,
```
Comment (skjnldsv) [inline webpack.config.js:63]: this is a breaking change @raimund-schluessler please revert

Comment (skjnldsv) [reply webpack.config.js:63]: meh, let me do that

Comment (skjnldsv) [reply webpack.config.js:63]: https://github.com/nextcloud/nextcloud-vue/pull/3644

File: webpack.config.js [hunk 3]
```diff
@@ -22,151 +24,108 @@ const SCOPE_VERSION = JSON.stringify(versionHash)
...
-		path: path.resolve(__dirname, './dist'),
-		publicPath: '/dist/',
-		filename: '[name].js',
-		library: {
-			type: libraryTarget,
-			name: ['NextcloudVue', '[name]'],
-		},
-		umdNamedDefine: true,
+webpackConfig.entry = {
+	index: path.join(__dirname, 'src', 'index.js'),
```
Comment (raimund-schluessler) [inline webpack.config.js:28]: @skjnldsv Comparing https://github.com/nextcloud/nextcloud-vue/blob/master/webpack.config.js with https://github.com/nextcloud/nextcloud-vue/blob/v7.3.0/webpack.config.js I think this also needs to go into the umd section here:
https://github.com/nextcloud/nextcloud-vue/blob/e5999882e2ae4fe22b2c72e3671ef509e1e8a36e/webpack.config.js#L159-L161

File: webpack.config.js (hunk 2 above)
Comment (skjnldsv) [reply webpack.config.js:63]: No worries at all :)

**Pull request merged**